src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,178 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
 #endif
 }
 
+TEST_CASE("Soroban Prefetch", "[ledgertxn]")
+{
+    VirtualClock clock;
+    auto cfg = getTestConfig(0, Config::TESTDB_DEFAULT);
+    cfg.ENTRY_CACHE_SIZE = 1000;
+    cfg.PREFETCH_BATCH_SIZE = cfg.ENTRY_CACHE_SIZE / 10;
+    cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+
+    UnorderedSet<LedgerKey> keysToPrefetch;
+    auto app = createTestApplication(clock, cfg);
+
+    auto& root = app->getLedgerTxnRoot();
+
+    auto entries = LedgerTestUtils::generateValidUniqueLedgerEntriesWithTypes(
+        {CONTRACT_CODE, CONTRACT_DATA}, cfg.ENTRY_CACHE_SIZE + 1);
+    std::set<LedgerEntry> entrySet;
+    std::map<LedgerKey, LedgerEntry> entryMap;
+    LedgerTxn ltx(root);
+    auto lkMeter = std::make_shared<LedgerKeyMeter>();
+    auto lkMeter2 = std::make_shared<LedgerKeyMeter>();
+    LedgerKeySet expectedSuccessKeys;
+    LedgerKeySet expectedFailedKeys;
+    for (auto e : entries)
+    {
+        ltx.createWithoutLoading(e);
+        keysToPrefetch.emplace(LedgerEntryKey(e));
+        UnorderedSet<LedgerKey> ks{};
+        ks.emplace(LedgerEntryKey(e));
+        stellar::uniform_int_distribution<uint32_t> dist(1, 10);
+        int32_t offset = 5 - dist(stellar::gRandomEngine);
+        lkMeter->addTxn(xdr::xdr_size(e) + offset, ks);
+        if (offset < 0)
+        {
+            REQUIRE(!lkMeter->canLoad(LedgerEntryKey(e), xdr::xdr_size(e)));
+            expectedFailedKeys.emplace(LedgerEntryKey(e));
+        }
+        else
+        {
+            REQUIRE(lkMeter->canLoad(LedgerEntryKey(e), xdr::xdr_size(e)));
+            expectedSuccessKeys.emplace(LedgerEntryKey(e));
+        }
+
+        // Insert entry into set with correct lastModifiedLedgerSeq value so
+        // we can check prefetch results later
+        e.lastModifiedLedgerSeq = 1;
+        entrySet.emplace(e);
+        entryMap.emplace(LedgerEntryKey(e), e);
+    }
+    if (cfg.EXPERIMENTAL_BUCKETLIST_DB)
```

Comment:
We can remove this if check since `cfg.EXPERIMENTAL_BUCKETLIST_DB == true` always for this test.